six/modules/c++/six.sidd/source/Utilities.cpp

```diff
@@ -1,615 +1,642 @@
+            }
+        */
+
+    default:
+        throw except::Exception(Ctxt("Invalid projection type: " +
+                                     p->projectionType.toString()));
+    }
+
+    return transform;
+}
+
+std::auto_ptr<scene::GridGeometry> Utilities::getGridGeometry(
+        const DerivedData* derived)
+{
+    if (!derived->measurement->projection->isMeasurable())
+    {
+        throw except::Exception(Ctxt(
+                "Projection type is not measurable: " +
+                derived->measurement->projection->projectionType.toString()));
+    }
+
+    const six::sidd::MeasurableProjection* p =
+            reinterpret_cast<const six::sidd::MeasurableProjection*>(
+                    derived->measurement->projection.get());
+
+    std::auto_ptr<scene::GridGeometry> geom;
+
+    // Only currently have an implementation for PGD
+    switch ((int)p->projectionType)
+    {
+    case six::ProjectionType::PLANE:
+    {
+        const six::sidd::PlaneProjection* const planeP =
+                reinterpret_cast<const six::sidd::PlaneProjection*>(p);
+
+        geom.reset(new scene::PlanarGridGeometry(
+                planeP->productPlane.rowUnitVector,
+                planeP->productPlane.colUnitVector,
+                p->referencePoint.ecef,
+                derived->measurement->arpPoly,
+                p->timeCOAPoly));
+        break;
+    }
+
+    default:
+        throw except::Exception(Ctxt("Invalid/unsupported projection type: " +
+                                     p->projectionType.toString()));
+    }
+
+    return geom;
+}
+
+void Utilities::setProductValues(Poly2D timeCOAPoly,
+                                 PolyXYZ arpPoly,
+                                 ReferencePoint ref,
+                                 const Vector3* row,
+                                 const Vector3* col,
+                                 types::RgAz<double> res,
+                                 Product* product)
+{
+    const double scpTime = timeCOAPoly(0, 0);
+
+    Vector3 arpPos = arpPoly(scpTime);
+    PolyXYZ arpVelPoly = arpPoly.derivative();
+    Vector3 arpVel = arpVelPoly(scpTime);
+
+    setProductValues(arpVel, arpPos, ref.ecef, row, col, res, product);
+}
+
+void Utilities::setProductValues(Vector3 arpVel,
+                                 Vector3 arpPos,
+                                 Vector3 refPos,
+                                 const Vector3* row,
+                                 const Vector3* col,
+                                 types::RgAz<double> res,
+                                 Product* product)
+{
+    const scene::SceneGeometry sceneGeom(arpVel, arpPos, refPos, *row, *col);
+
+    // do some setup of derived data from geometry
+    if (product->north == Init::undefined<double>())
+    {
+        product->north = sceneGeom.getNorthAngle();
+    }
+
+    // if (product->resolution
+    //    == Init::undefined<RowColDouble>())
+    {
```

Comment:
The if statement is commented out, should this whole block be commented out?

Reply:
Good question.  Looks like this has been commented out for the entire available Git history, so I think the best thing to do is to keep not touching it. When Bill gets back, I can see if he has any idea what's going on here.